### PR TITLE
ci: Turn off builds for pushes and merges to master/stable* and cancel parallel executions

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

**Review by provokateurin — APPROVED** [review]

**Review by SystemKeeper — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

File: .github/workflows/node-tests.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@ name: Node tests
 
 on:
   pull_request:
-  push:
```
Comment (susnux) [inline .github/workflows/node-tests.yml:5]: This will disable code coverage

**Review by susnux — COMMENTED** [review]
This is needed and good, but we should keep node-tests and (later) phpunit tests run on push to master & stable* because otherwise the project code coverage can not be calculated.

There will only be partially values for PRs and if they only touch JS then only the JS coverage will be calculated.

But I think we should also calculate a overall project coverage to see whats missing to be tested, therefor we should run tests that count into code coverage on pushes to, at least, master.

File: .github/workflows/node-tests.yml (hunk 1 above)
Comment (SystemKeeper) [reply .github/workflows/node-tests.yml:5]: Could we do this once a day maybe or does it need to happen on push?

Comment (nickvergessen) [reply .github/workflows/node-tests.yml]: reverted

**Comment by nickvergessen** [comment]
> But I think we should also calculate a overall project coverage to see whats missing to be tested, therefor we should run tests that count into code coverage on pushes to, at least, master.

Reverted for now.

We could also just do it every night. I guess no one will look at it "live", so that should be totally fine and would help to reduce the run times even more.

**Comment by susnux** [comment]
> We could also just do it every night. I guess no one will look at it "live", so that should be totally fine and would help to reduce the run times even more.

This sounds good!

**Review by st3iny — APPROVED** [review]

File: .github/workflows/node-tests.yml (hunk 1 above)
Comment (susnux) [reply .github/workflows/node-tests.yml:5]: > Could we do this once a day maybe or does it need to happen on push?

No it just needs to be done on the correct branch

**Pull request merged**